src/pages/storage/panels/EditStorageBucketPanel.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  ActionButton,
+  Button,
+  useNotify,
+  useToastNotification,
+} from "@canonical/react-components";
+import SidePanel from "components/SidePanel";
+import type { FC } from "react";
+import usePanelParams from "util/usePanelParams";
+import { useFormik } from "formik";
+import NotificationRow from "components/NotificationRow";
+import ScrollableContainer from "components/ScrollableContainer";
+import ResourceLink from "components/ResourceLink";
+import StorageBucketForm from "../forms/StorageBucketForm";
+import type { StorageBucketFormValues } from "../forms/StorageBucketForm";
+import { updateStorageBucket } from "api/storage-buckets";
+import { queryKeys } from "util/queryKeys";
+import { useQueryClient } from "@tanstack/react-query";
+import type { LxdStorageBucket } from "types/storage";
+import { pluralize } from "util/instanceBulkActions";
+
+interface Props {
+  bucket: LxdStorageBucket;
+}
+const EditStorageBucketPanel: FC<Props> = ({ bucket }) => {
+  const panelParams = usePanelParams();
+  const notify = useNotify();
+  const toastNotify = useToastNotification();
+  const queryClient = useQueryClient();
+  const closePanel = () => {
+    panelParams.clear();
+    notify.clear();
+  };
+
+  const handleSuccess = (bucketName: string) => {
+    toastNotify.success(
+      <>
+        Bucket{" "}
+        <ResourceLink
+          type="bucket"
+          value={bucketName}
+          to={`/ui/project/${panelParams.project}/storage/buckets`}
+        />{" "}
+        updated.
+      </>,
+    );
+    closePanel();
+  };
+
+  const formik = useFormik<StorageBucketFormValues>({
+    initialValues: {
+      name: bucket?.name,
+      pool: bucket?.pool,
+      size: bucket?.config.size,
+      description: bucket?.description,
+      target: bucket?.location,
+    },
+    enableReinitialize: true,
+    onSubmit: (values) => {
+      const bucketPayload = {
+        name: values.name,
+        config: { size: values.size },
+        description: values.description,
+      } as LxdStorageBucket;
+
+      updateStorageBucket(
+        bucketPayload,
+        values.pool,
+        panelParams.project,
```

Comment:
I think we removed the project from the panel params. This should use the current project.